packages/account-export/src/lib/helpers/importAccount.ts

```diff
@@ -74,12 +73,7 @@ export const decryptAccountData = ({
   }
 };
 
-export const generateSecretKey = (): string =>
-  generator.generate({
-    length: nacl.secretbox.keyLength,
-    numbers: true,
-    strict: true,
-    lowercase: true,
-    uppercase: true,
-    symbols: true,
-  });
+export const generateSecretKey = (): string => {
+  const random = nacl.randomBytes(nacl.secretbox.keyLength);
+  return encodeBase64(random).substring(0, 32);
```

Comment:
@andy-haynes I would like to get your feedback on here!
While I spent time on using `nacl` to generate 32 character secret key, I couldn't figure out and  ended up using `substring` to enforce the character size to 32. Do you know more elegant solution on this?

Reply:
The byte array contains wider (2^8) values than base 64 (2^6), so encoding in base 64 requires more symbols to represent the same amount of information. The length of the encoded string is always 44, with the last character being `=` for padding.  So the byte array of length 32 represents up to `(2^8)^32 = 256^32 = 1.2e77` values and a base 64 string of length 43 is `(2^6)^43 = 64^43 = 4.6e77`, making 43 the lowest number of base 64 characters you can use to represent all values in the byte array.

To get a 32 character base 64 string, you need a byte array of length `log_256(64^32) = 24` (https://www.wolframalpha.com/input?i=log+base+256+%2864%5E32%29). We should be able to rely on that to always produce strings of length 32 😄

Reply:
Also the byte array of length 32 always had a `=` padding character because 32 is not a multiple of 3. A byte array of length 24 is a multiple of 3 and so will never contain padding, so we will always make the most of our 32 characters.

https://en.wikipedia.org/wiki/Base64#Output_padding

> Because Base64 is a six-bit encoding, and because the decoded values are divided into 8-bit octets, every four characters of Base64-encoded text (4 sextets = 4 × 6 = 24 bits) represents three octets of unencoded text or data (3 octets = 3 × 8 = 24 bits). This means that when the length of the unencoded input is not a multiple of three, the encoded output must have padding added so that its length is a multiple of four. The padding character is =, which indicates that no further bits are needed to fully encode the input.

Reply:
@andy-haynes awesome! thank you for guidance and explanation!